mulog-kinesis/src/com/brunobonacci/mulog/publishers/kinesis.clj

```diff
@@ -0,0 +1,70 @@
+(ns com.brunobonacci.mulog.publishers.kinesis
+  (:require [com.brunobonacci.mulog.buffer :as rb]
+            [com.brunobonacci.mulog.utils :as ut]
+            [com.brunobonacci.mulog.publishers.aws-utils :as awsutils]
+            [cheshire.core :as json]
+            [cheshire.generate :as gen]))
+
+;;
+;; Add Flake encoder to JSON generator
+;;
+(gen/add-encoder com.brunobonacci.mulog.core.Flake
+                 (fn [x ^com.fasterxml.jackson.core.JsonGenerator json]
+                   (gen/write-string json ^String (str x))))
+
+(defn- put-records
+  [{:keys [stream-name partition-key-name format max-items] :as config} records]
+  (let [key-field partition-key-name
+        fmt* (if (= :json format) json/generate-string ut/edn-str)]
+    (->> records
+         (map (juxt #(str (get % key-field)) fmt*))
+         (map (fn [[k v]] (awsutils/create-records! k v)))
+         (partition-all max-items)
```

Comment:
no need to do this, it will be managed by the publisher
see the offset use in
https://github.com/BrunoBonacci/mulog/blame/master/doc/custom-publishers.md#L161-L164